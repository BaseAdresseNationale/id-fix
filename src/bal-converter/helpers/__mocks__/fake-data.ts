import { idsIdentifierIndex } from "../../bal-converter.config.js";

export const addrID = "aaaaaaaa-0000-4aaa-9000-1234567890aa";
export const mainTopoID = "bbbbbbbb-0000-4aaa-9000-1234567890bb";
export const secondaryTopoIDs = [
  "cccccccc-0000-4aaa-9000-1234567890cc",
  "cccccccc-1111-4aaa-9000-1234567890dd",
  "cccccccc-2222-4aaa-9000-1234567890ee",
];

const {
  addressID: addressIDKey,
  mainTopoID: mainTopoIDKey,
  secondaryTopoIDs: secondaryTopoIDsKey,
} = idsIdentifierIndex;

export const idSampleWithBanId = `${addressIDKey}${addrID}`;
export const idSampleWithMainTopoId = `${mainTopoIDKey}${mainTopoID}`;
export const idSampleWithSecondaryTopoId = `${secondaryTopoIDsKey}${secondaryTopoIDs.join(
  "|"
)}`;
export const idSampleWithBanIdAndMainTopoId = `${addressIDKey}${addrID} ${mainTopoIDKey}${mainTopoID}`;
export const idSampleWithBanIdAndSecondaryTopoId = `${addressIDKey}${addrID} ${secondaryTopoIDsKey}${secondaryTopoIDs.join(
  "|"
)}`;
export const idSampleWithAllIds = `${addressIDKey}${addrID} ${mainTopoIDKey}${mainTopoID} ${secondaryTopoIDsKey}${secondaryTopoIDs.join(
  "|"
)}`;
