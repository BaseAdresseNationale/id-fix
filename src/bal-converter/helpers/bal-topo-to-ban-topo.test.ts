import type { BalAdresse } from "../../types/bal-types.js";
import type { BanCommonToponym } from "../../types/ban-types.js";

import { describe, expect, test } from "vitest";
import { idsIdentifierIndex } from "../bal-converter.config.js";
import { addrID, mainTopoID, secondaryTopoIDs } from "./__mocks__/fake-data.js";
import balTopoToBanTopo from "./bal-topo-to-ban-topo";

const idSampleWithBanId = `${idsIdentifierIndex.addressID}${addrID}`;
const idSampleWithMainTopoId = `${idsIdentifierIndex.mainTopoID}${mainTopoID}`;
const idSampleWithBanIdAndMainTopoId = `${idsIdentifierIndex.addressID}${addrID} ${idsIdentifierIndex.mainTopoID}${mainTopoID}`;
const idSampleWithAllIds = `${idsIdentifierIndex.addressID}${addrID} ${
  idsIdentifierIndex.mainTopoID
}${mainTopoID} !${secondaryTopoIDs.join("|")}`;

const defaultTestBalAdresse: BalAdresse = {
  cle_interop: "21286_0001_00001",
  commune_insee: "21286",
  commune_nom: "Cocorico",
  voie_nom: "Route de la Baleine",
  numero: 1,
  position: "autre",
  x: 1,
  y: 2,
  long: 1,
  lat: 2,
  date_der_maj: "2021-01-01",
  certification_commune: true,
  source: "BAL",
};

describe("balTopoToBanTopo", () => {
  test("should return BanToponym without BanTopoID", async () => {
    expect(balTopoToBanTopo(defaultTestBalAdresse)).toMatchSnapshot();
  });

  test("should return BanToponym without BanTopoID", async () => {
    // TODO: In next version - should probably return BanToponym without temporary BanTopoID ?
    const testBalAdresse: BalAdresse = {
      ...defaultTestBalAdresse,
      uid_adresse: idSampleWithBanId,
    };
    expect(balTopoToBanTopo(testBalAdresse)).toMatchSnapshot();
  });

  test("should return BanToponym with BanTopoID (1)", async () => {
    const testBalAdresse: BalAdresse = {
      ...defaultTestBalAdresse,
      uid_adresse: idSampleWithMainTopoId,
    };
    expect(balTopoToBanTopo(testBalAdresse)).toMatchSnapshot();
  });

  test("should return BanToponym with BanTopoID (2)", async () => {
    const testBalAdresse: BalAdresse = {
      ...defaultTestBalAdresse,
      uid_adresse: idSampleWithBanIdAndMainTopoId,
    };
    expect(balTopoToBanTopo(testBalAdresse)).toMatchSnapshot();
  });

  test("should return BanToponym with BanTopoID and other toponyms", async () => {
    const testBalAdresse: BalAdresse = {
      ...defaultTestBalAdresse,
      uid_adresse: idSampleWithAllIds,
    };
    expect(balTopoToBanTopo(testBalAdresse)).toMatchSnapshot();
  });

  test("should return BanToponym with multilingual label", async () => {
    const testBalAdresse: BalAdresse = {
      ...defaultTestBalAdresse,
      uid_adresse: idSampleWithMainTopoId,
      voie_nom: "Route de la Baleine",
      voie_nom_eus: "Baleen ibilbidea",
    };
    expect(balTopoToBanTopo(testBalAdresse)).toMatchSnapshot();
  });

  test("should return BanToponym with overwrited data", async () => {
    // TODO : In next version - should probably not return BanToponym with overwrited data and throw an error ?

    const testOldBalAdresse: BalAdresse = {
      ...defaultTestBalAdresse,
      uid_adresse: idSampleWithAllIds,
      voie_nom: "Route de la Baleine",
    };
    const oldBanToponym = balTopoToBanTopo(
      testOldBalAdresse
    ) as BanCommonToponym;

    const testBalAdresse: BalAdresse = {
      ...defaultTestBalAdresse,
      uid_adresse: idSampleWithAllIds,
      voie_nom: "Avenue Rhoam Bosphoramus",
    };
    expect(balTopoToBanTopo(testBalAdresse, oldBanToponym)).toMatchSnapshot();
  });
});
