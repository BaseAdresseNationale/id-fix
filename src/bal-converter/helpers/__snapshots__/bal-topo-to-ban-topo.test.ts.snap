// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`balTopoToBanTopo > should return BanToponym with BanTopoID (1) 1`] = `
{
  "districtID": undefined,
  "id": "bbbbbbbb-0000-4aaa-9000-1234567890bb",
  "labels": [
    {
      "isoCode": "fra",
      "value": "Route de la Baleine",
    },
  ],
  "meta": {
    "bal": {
      "deprecatedID": "21286_0001",
    },
  },
  "updateDate": 2021-01-01T00:00:00.000Z,
}
`;

exports[`balTopoToBanTopo > should return BanToponym with BanTopoID (2) 1`] = `
{
  "districtID": undefined,
  "id": "bbbbbbbb-0000-4aaa-9000-1234567890bb",
  "labels": [
    {
      "isoCode": "fra",
      "value": "Route de la Baleine",
    },
  ],
  "meta": {
    "bal": {
      "deprecatedID": "21286_0001",
    },
  },
  "updateDate": 2021-01-01T00:00:00.000Z,
}
`;

exports[`balTopoToBanTopo > should return BanToponym with BanTopoID and BanDistrictID 1`] = `
{
  "districtID": "dddddddd-0000-4aaa-9000-1234567890ff",
  "id": "bbbbbbbb-0000-4aaa-9000-1234567890bb",
  "labels": [
    {
      "isoCode": "fra",
      "value": "Route de la Baleine",
    },
  ],
  "meta": {
    "bal": {
      "deprecatedID": "21286_0001",
    },
  },
  "updateDate": 2021-01-01T00:00:00.000Z,
}
`;

exports[`balTopoToBanTopo > should return BanToponym with multilingual label 1`] = `
{
  "districtID": undefined,
  "id": "bbbbbbbb-0000-4aaa-9000-1234567890bb",
  "labels": [
    {
      "isoCode": "fra",
      "value": "Route de la Baleine",
    },
    {
      "isoCode": "eus",
      "value": "Baleen ibilbidea",
    },
  ],
  "meta": {
    "bal": {
      "deprecatedID": "21286_0001",
    },
  },
  "updateDate": 2021-01-01T00:00:00.000Z,
}
`;

exports[`balTopoToBanTopo > should return BanToponym with multilingual label after cleaning ISO code lang 1`] = `
{
  "districtID": undefined,
  "id": "bbbbbbbb-0000-4aaa-9000-1234567890bb",
  "labels": [
    {
      "isoCode": "fra",
      "value": "Baleen ibilbidea",
    },
    {
      "isoCode": "unknow",
      "value": "Anything",
    },
  ],
  "meta": {
    "bal": {
      "deprecatedID": "21286_0001",
    },
  },
  "updateDate": 2021-01-01T00:00:00.000Z,
}
`;

exports[`balTopoToBanTopo > should return BanToponym with multilingual label and ignore empty default value 1`] = `
{
  "districtID": undefined,
  "id": "bbbbbbbb-0000-4aaa-9000-1234567890bb",
  "labels": [
    {
      "isoCode": "fra",
      "value": "Baleen ibilbidea",
    },
  ],
  "meta": {
    "bal": {
      "deprecatedID": "21286_0001",
    },
  },
  "updateDate": 2021-01-01T00:00:00.000Z,
}
`;

exports[`balTopoToBanTopo > should return BanToponym with multilingual label and replace default value 1`] = `
{
  "districtID": undefined,
  "id": "bbbbbbbb-0000-4aaa-9000-1234567890bb",
  "labels": [
    {
      "isoCode": "fra",
      "value": "Route de la Baleine",
    },
  ],
  "meta": {
    "bal": {
      "deprecatedID": "21286_0001",
    },
  },
  "updateDate": 2021-01-01T00:00:00.000Z,
}
`;

exports[`balTopoToBanTopo > should return BanToponym with overwrited data 1`] = `
{
  "districtID": "dddddddd-0000-4aaa-9000-1234567890ff",
  "id": "bbbbbbbb-0000-4aaa-9000-1234567890bb",
  "labels": [
    {
      "isoCode": "fra",
      "value": "Avenue Rhoam Bosphoramus",
    },
  ],
  "meta": {
    "bal": {
      "deprecatedID": "21286_0001",
    },
  },
  "updateDate": 2021-01-01T00:00:00.000Z,
}
`;

exports[`balTopoToBanTopo > should return BanToponym without BanTopoID 1`] = `
{
  "districtID": undefined,
  "id": undefined,
  "labels": [
    {
      "isoCode": "fra",
      "value": "Route de la Baleine",
    },
  ],
  "meta": {
    "bal": {
      "deprecatedID": "21286_0001",
    },
  },
  "updateDate": 2021-01-01T00:00:00.000Z,
}
`;

exports[`balTopoToBanTopo > should return BanToponym without BanTopoID 2`] = `
{
  "districtID": undefined,
  "id": undefined,
  "labels": [
    {
      "isoCode": "fra",
      "value": "Route de la Baleine",
    },
  ],
  "meta": {
    "bal": {
      "deprecatedID": "21286_0001",
    },
  },
  "updateDate": 2021-01-01T00:00:00.000Z,
}
`;
